src/js/select2/core.js

```diff
@@ -473,7 +490,7 @@ define([
    * @return {false} if the disabled option is false.
    */
   Select2.prototype.isDisabled = function () {
-    return this.options.get('disabled');
+    return this.options.get('disabled') || this.$element.is(':disabled');
```

Comment:
Two things here.

1. The expected behaviour is to sync the "disabled" option with the state of the element. We do that already but we only check the `disabled` property.

  https://github.com/select2/select2/blob/ef33f4be110f5154c11b49d73e7b15921e143398/src/js/select2/core.js#L339

2. We do not currently use the `:disabled` jQuery selector because it relies on the Sizzle engine. If you are able to replace the `.is()` call with a vanilla JS call then we are more likely to be able to work that fix into Select2.